### ISSUE TITLE
🗒 Licensing

### ISSUE THREAD

**Issue by alice-i-cecile** [katharostech/bevy_retro#1]
I noticed there was no LICENSE.md on this repo, which makes it very challenging for others to build off of safely.

This work looks *incredible*, and I (among many others) would love to be able to use it, contribute to it, promote it and so on. To do so though, it needs a reasonably standard open source license.

The Katharos license used on your other projects, independent of any planned uses or ethical / religious disagreements, is a complete deal breaker for every interested user I've talked to :( It induces very complex legal risk due to challenges in definition and lack of case law, and even incorporating a library with the license presents a serious barrier to getting downstream users and contributors who may disagree with certain ethical or religious implications contained within the license.

Promoting good in the world and staying true to your beliefs is the right thing to do, and I admire your devotion to what you believe in. I'm just not convinced that ethical open source licenses are a good tool to get there. 

The intent behind them is twofold:
1. Prevent your work from being used for objectionable purposes.
2. Allow others who share your goals to benefit from your work.

But in effect, the level of uncertainty induced by them is such that only the first goal succeeds at all, leaving these fantastic projects that can't be used by *anyone* completely stranded. 

Feel free to reach out, either here or by email. I'm happy to talk at length about my perspective on these matters, and see if I can help you come to a solution that helps you share and collaborate in a way that's compatible with your values.

**Comment by zicklag** [comment]
Hi @alice-i-cecile, thanks for reaching out! The lack of a LICENSE.md was accidental, but as you guessed the Katharos License was the license that we would be releasing this under.

I definitely see your point about the uncertainty and potential legal risk involved in building on top of our license. I believe your concerns there are completely valid. This is something that I will bring up with my team, and we may take you up on your offer to discuss. I will let you know as soon as I can get a concensus with my team.

Thanks again for posting. :slightly_smiling_face: :+1:

**Comment by zicklag** [comment]
Sent you an Email @alice-i-cecile. For any watchers, I'll comment here with whatever conclusion we come to.

**Comment by konsumer** [comment]
I hope I am not out-of-place jumping in, but I got very excited about bevy_retro, then saw the license, and had to rethink it. Maybe I can share an end-user (and future-contributor) perspective. If any of this is offensive, please know that is very much not the spirit of my comment. I am unused to talking about religion in this context, so please try to hear the other things I have to say, as I am very new to talking about it in this setting, and mean no offense.

bevy_retro is really seriously solid work, and does exactly what I need, I think. The other related projects, (skipngo & LDtk loader) are just amazing, and align perfectly with what I want to do, written in very clear code, that seems to work really well. A few things about the license would keep me from contributing much, promoting, or basing any future work on it, though:

- I am an atheist. Everyone interprets "God's will" differently. Some don't even have a God, like me, but we still can have overlapping goals or help each other get awesome things made. An example: I bet there is at least 1 Bevy contributor that is an atheist or Muslim. Even though they don't have the same religion, you still have the common goal of making a software library to make rad things, and without their work you couldn't make bevy_retro. This goes all the way back to the silicone in the CPU. I guarantee there were engineers with completely different morals from me & you (some even in direct opposition) in the many people that helped make this computer I am typing on. If one guy on the CPU-team at Intel said "I only allow my silicone layout design to be in the service of Alah" how would that make you feel?
- I have seen "the will of God" used by others to do/justify things I am very strongly morally opposed to (like war and hate-crimes, for example.) It makes me extremely wary of any other use of this, and I also don't want to pass that burden down to users of my stuff
- My gay & trans colleagues, friends and family have often been shunned/actively-hurt by Christians, and have very bad associations with that. These are some of the most important people in my life, so I wouldn't want to give them the impression I am ok with that, or associated with people that have made their lives so much harder.
- Similarly, many of my colleagues, friends and family are in direct religious opposition to Christianity. Atheists, Muslims, etc, and they are all very important to me. I'm ok with them not having the same religion as me (none) and I want them to know I am ok with their religions, too. I support you being a Christian with a strong moral compass, as long as it doesn't infringe on my moral compass, and I expect the same from everyone I work with.
- In my current game, I will allow users to make their own mini-games, and I don't want to put that content burden on them, especially if it's a 3rd party set of not-totally-clear morals. As I said, people interpret the bible very differently from each other, and I'd rather not leave it up to chance that our morals, and all morals of any contributors or users of my code, will stay in line. They might not, and personally I'm ok with that.
- It could be completely against other terms of service (like Github's) to have a religion-based moral clause in your code, and I don't want my own projects to have the same legal risk.

I think I get the vibe behind the license, and I do believe your heart is in the right place, from what I can tell. I'm not even in total disagreement with some of what you might mean by it, even though I couldn't agree to the license, as it is. Here are some ideas that would make it much easier for me to accept, and work with/contribute/promote your projects:

- Just a regular conventional open-source license. This is probably the easiest for the most people the use. Think of the license as just protection from people using your actual code in a way you don't want, like selling it without sharing source, etc. There are high-precedent licenses (MIT, GPL2/3, etc) and you (and your down-stream users like me) will get a lot of free legal protection against people violating these licenses. I am totally unsure if your license is enforceable in the first place, which would mean that other parts that are also really important to everyone, are also not enforceable, exposing me, and my downstream users, to a great legal risk 
- Put moral stuff in a code of conduct instead, and try to leave out the stuff about God & the bible, in order to make it more inclusive. Think of the actual specific use-cases you are opposed to. I am totally fine with "no gory games or games that promote war in my issue queues" but I am not ok with "no one can ever use this library, no matter how far downstream, to make gory war games"
- Maybe think about it more as a clause for what your company promotes, instead of how the library can be used at all. As more people use your library (and a more inclusive license will definitely help with that) you will have more of a platform. Use that platform to amplify projects that agree with your morals, and you can essentially get the same effect, without keeping anyone out that disagrees with other parts of your beliefs.
- I am actually a huge fan of "not for war" moral clause, not just because it aligns with my morals (I hate war and don't want to contribute software to it) but because it is pretty specific, without requiring end-users to agree with me on other things. They just have to promise not to make missile-guidance software, not be "guided by the bible". That seems specific enough that is it actionable and clear, and at this point, even has a legal precedent.

Hopefully that all makes sense and adds something to the discussion. Even if you don't change the license, I wish you well, and am seriously impressed with the work that you've done.

**Comment by zicklag** [comment]
@konsumer, you aren't out of place and thank you for your perspective.

---

I want to give a good picture of me and my organization's perspective on this and why we made the license what it is. This is, frankly, a religious explanation, but that is because our faith is why me, and my organization, are here. Our religion is _why_ we write code, or do anything else, so it is crucial to this issue. I speak not necessarily for everybody on my team, but for me and the founders who chose to make Katharos Technology and who choose it's direction as a company.

My company, [Katharos Technology](https://katharostech.com/about-us), is founded on the concept that we are made by God to do His will. We would prefer that our company be abolished and dissolved rather than have our company serve the will of man above the will of God. If we are not serving God with our company, then we have undermined the reason for its existence. This is the core principal of our company and everything we do is built on that principle.

We believe that God has led us to make Video Games. We, personally, have a passion for Video Games and the wonder and fun that they can bring. It is something that we have enjoyed for a long time and we feel that the modern market is severely lacking in good, clean Video Games that we are comfortable exposing ourselves to. We believe that we can share good and uplifting games and be a blessing to others through it, and that God has given us the talents, time, and guidance to we need to do so. Our work to make games is based on the premise that we believe it is work that God wants us doing.

We are also passionate about Open Source. Our company is built on Open Source tooling and we enjoy much of the community and spirit behind the Open Source community and tools that we use. We honestly desire to give the community as much of what we have as we can. We really do _want_ to share what we have made and give other people the ability to use it.

But...our first priority is to do God's will.

If you consider the Video Games in the world today, you will find that the large majority of them contain content that is against the whole purpose for our company. These games contain content that we believe is damaging to people. We want our contributions and efforts to be uplifting, not damaging.

All of our game engines and libraries are _specifically_ designed to make it as easy as absolutely possible to make games, _especially_ for those who would otherwise not be able to because of the difficulty involved. But if we make it easier for people to make _any kind of game_ we are going to be directly contributing to games that are contrary to our company's reason for existing. We will, in all likelyhood, be contributing more to what goes against our purpose than to what is for our purpose, and we would sooner make it 100% proprietary than do that.

Our license is an attempt at solving this problem, but it is by no means perfect. As @alice-i-cecile said, it successfully accomplishes preventing objectionable use, but it may not successfully allow us to share our work with the community. The legal concerns that she brings up are valid concerns and I don't argue that they aren't, but if our license means that the only people who can use our tools are a subset of hobbyists, then that is still better than us 100% proprietizing it.

Our license was not made to try and keep things to ourselves for our own advantage. If there was a way for us to give free use of our software for anything that does not propagate what we believe is harmful to people, regardless of their religion or what they believe, then we would gladly do that. As it stands, though, we don't know what better options we have and this license is the closest thing we know to accomplishing it.

That is the _reason_ behind the license and why we are steadfast in supporting it over typical Open Source licenses. It's not just about the license, it's about why we are here as a company.

---

On a more technical note, I also want to address a common misconception about the license: all of the text in the "Preamble" section of the license has no bearing on the actual terms of the license, the legal rules that you must follow.

This is very similar to the [preamble](https://www.gnu.org/licenses/gpl-3.0.html#preamble) of the GPL license, which has a "human" English explanation of the purpose of the license. The terms of the Katharos License don't actually make reference to God. ( It does make one mention of the Bible in case of a conflict with human's rights laws and the permitted use of work, but we have removed that in the next draft because of it's ambiguity. ) This means that the license does not require an interpretation of "God's will". We don't require you to believe what we believe, just that you follow the terms outlined in the license.

That said, the terms as they are can also be ambiguous. This is something we are working on helping with [version 0.2](https://github.com/katharostech/katharos-license/blob/e6d4abe2b5a7a9584cc080176da3d0a31488866b/LICENSE_v0.2.0_draft.md) of the license. With the second draft of the license we tried to clean up as much ambiguity as we could, but it does not solve the lack of case law or questions of enforceability. Also it maintains the same spirit and applies to all downstream dependents regardless of depth.

---

I hope that helps explain our side of things. I appreciate your thought-out responses, and your interest in our work, but unless a better solution presents itself, our license will remain and we honestly believe that this is the will of our God and that it is what He wants us to do.

If this means that many will not be able to use or even contribute and help promote our tools, we still hold that we believe this is the decision that God wants us to make and that He will bless us for following His will above our own.

**Comment by konsumer** [comment]
That does make sense, and thanks for clarifying. As it is, I don't think I will be able to use any of this stuff, but I understand where you are coming from, I think. If you could use another perspective in future discussion about this, don't hesitate to reach out.

**Comment by erlend-sh** [comment]
I want to point out how great it is to see everyone entering into this discussion with grace and kindness, in spite of how contentious fraught with emotion the topic of religion can be.

Because we’re used to open source practices being a certain way, I think there’s a framing missing from this conversation that can help set it in a different light.

I also don’t subscribe to any specific religion, so some of the language of this license is quite foreign to me. However, I find the basic terms to be pretty straight forward. Rather than looking at it through an open source lens, I think of it instead as a traditional agreement for proprietary software, like paying for a Photoshop subscription:

1. Pay (comply with) the license fee.
2. Enjoy access to the software as long as you comply with the business terms.

How do I comply with the Katharos license? Well I’m basically limited to making [PEGI 12](https://pegi.info/what-do-the-labels-mean) games. Those are constraints I can easily work within.

Furthermore, the uncertainty of the license’s future (Will it change? Will it hold up in court?) is no different from a closed tool that I might be using for free as long as it’s in beta. The company behind this beta product could disappear, or they might introduce pricing that is way outside my budget. 

My use of such a tool would come down to a very simple evaluation of time cost: Is this tool saving me more time now, than the time it might cost me in the future to replace said tool?

Seeing as the Katharos toolset is source-available ([much lower lock-in than the average proprietary software](https://openeveryone.substack.com/p/on-the-spectrum-of-openness)) and built on top of the fully open source Bevy, I’m not very concerned that any game I make with Katharos tooling is gonna get trapped in some license dispute. I can always go with the nuclear option and rewrite whatever’s necessary, but I see no reason to do that as long as I’m keeping an open line of dialogue with Katharos team.

With that in mind, I would suggest one addition to the license: A clearly defined grace period for non-compliance. If my project is deemed not to be in compliance with the license, how long do I have to either (a) become compliant or (b) stop using the licensed software?

**In summary:** By **faaar** the hardest part of any game project is just getting a game to some reasonable version of Done. Gamedevs scrap and rewrite prototype code more than any other software discipline. No piece of tooling we use is safe from replacement. If some tool gets you considerably closer to the game you wanna make *today*, most likely I’m gonna recommend that you use that tool. A month or two from now you’re not gonna be developing that game anymore if you spent all your early momentum building tools “for the longterm”.

**Comment by zicklag** [comment]
Thanks for chiming in @erlend-sh! I think you bring a very useful perspective to this.

> A clearly defined grace period for non-compliance. 

That's a great idea. We can make sure that that goes into the next draft. :+1:

**Comment by leeola** [comment]
I am also very concerned about this license. I'd really need the input from legal council on this license. I suspect that adhering to a rule set as defined by 66 books of some of the most widely debated text in human history is a massive legal liability. Furthermore, the license does not even define _which specific_ version/translation of the widely debated text is to be used in court.

As i _(not a lawyer and know little on licensing)_ see it, either KatharosTech has ultimate say in what is allowed or not allowed based on their interpretation of whatever Bible they choose, or the full work of some unnamed version of the Bible is to be debated in court. If it's the former, having KatharosTech having ultimate say would seem to effectively make this license dependent on Katharos's opinion _(very subjective, counter to what the License claims)_. If it's the latter, i can't even begin to imagine a court case trying to prove someone to be in violation of a Bible.

I mean no disrespect towards your faith, but i can't imagine how this will fare in a legal context. Would _any_ lawyer even recommend a company use this library, based on this license? It screams of liability to me. I can't even envision what or how it would be enforced. I do wonder if this license, which seems to simply defer to an unnamed version of the Bible, has effectively become the longest open source license to date. Perhaps not, but curious nonetheless.

To summarize; I personally would require legal council to even use this library to assess the risk.

**Comment by zicklag** [comment]
> I'd really need the input from legal council on this license. I suspect that adhering to a rule set as defined by 66 books of some of the most widely debated text in human history is a massive legal liability. 

The mention of The Bible in the prelude of the license does not have any bearing on the legal terms of the license. The terms contain only a single reference to The Bible in an edge case regarding a conflict with local humans rights laws, and that reference _is_ being removed in the next version of the license.

The reference to The Bible in the prelude was to indicate our stance on why we felt that there should be restrictions and our outlook on what defines what "right" is. We believe the bible defines what is right, but because that is subjective and, as you say, unreasonable to use as the foundation for legal terms directly, we defined the exact terms of the license _after_ the prelude. Those terms below are the only terms you are bound to.

As to the content of those terms they are derived almost fully from the [hippocratic license](https://firstdonoharm.dev/) but with the humans rights laws being replaced with our list of what we don't want you using the software for. For what it is worth, we did not design the legalese of this license fully ourselves, but built on a pre-existing ethical source license and swapped out the terms of use with what we are concerned with.

**Comment by erlend-sh** [comment]
I think at this point we’re just rehashing old remarks. Probably fine to close as ‘answered’?

**Comment by zicklag** [comment]
Sounds good. :+1:

**Comment by felixwatts** [comment]
> The mention of The Bible in the prelude of the license does not have any bearing on the legal terms of the license

I disagree.

> The Katharos License is based on the premise that the full 66 books of the Holy Bible are 100% true

The 66 books of the bible contain [many internal contradictions](https://en.wikipedia.org/wiki/Internal_consistency_of_the_Bible) and cannot, by definition, all be true. This fact has been well documented [for centuries](https://en.wikipedia.org/wiki/The_Age_of_Reason#:~:text=The%20Age%20of%20Reason%3B%20Being,the%20philosophical%20position%20of%20deism.).

Being based on a premise that is demonstrably false does not seem like a good starting point for what aims to be a legally sound document.

**Comment by ColonelThirtyTwo** [comment]
I hope you don't advertise your code as open-source, because this definitely does not meet the [OSS criterion](https://opensource.org/osd-annotated).

> * sexually suggestive or explicit images, artwork, or any other media
> * excessively gory and/or violent images, artwork, or any other media

Both of these restrictions are too vague. At what point does a game become "suggestive" or "excessively" gory? The restriction on sexuality is also pretty prudish.

In any case, I can't use or support this project with the license as it is - I realize you mean well, but there is simply too much risk.

**Comment by SonicZentropy** [comment]
I know this was already closed once and the discussion is pretty much over, but I wanted to commend you (as someone who would like to use this repo but cannot due to the license) for being extremely open and up-front about the licensing everywhere.  Although we disagree, I appreciate your respectful and open approach.  I do wish I could use it though!

**Comment by azbay** [comment]
As others have mentioned, the issue is obviously closed, but I'm still baffled that this is even a thing. Others have been far too kind in their responses, and while I appreciate common courtesy online, it really sucks to see an otherwise useful asset be tarnished by such a stupid and thoughtless idea.

As already discussed, I'm aware that the license itself (aside from the Preamble) doesn't rely on the Christian Bible, but the inspiration/reasoning here is silly at best.

`The definition of what is "good" can be considered highly subjective. In order to maintain objectivity in a highly subjective matter, there must be some source of truth from which to derive said objectivity. The source of "truth" for the Katharos License, and where the definition of what is "good" and "pure", come from the Word of God, The Holy Bible. The Katharos License is based on the premise that the full 66 books of the Holy Bible are 100% true and inspired by God and that He alone is the ultimate authority for what is good and just.`

This doesn't even make sense from a Christian perspective. There are some 45,000 denominations of Christianity, so there's at least one hundred times as many different interpretations of the verses of the Bible - most of which are based on certain translations, of which there are thousands. To claim you understand any "objective" moral good from the Bible is to claim that you can somehow invoke God's will directly, or that you perfectly understand His intent when divinely inspiring (or writing directly? I won't presume to know your stance) the text. Alternatively, you're claiming that _your interpretation_ of the Bible is somehow objective truth from which you can derive a legal document that will be useful to others, which is equally as preposterous. Wars and rebellions have been fought over more mild theological claims than that.

`The Work shall not be Used by any person or entity to display, market, or distribute images, artwork, or any other media that is:
- sexually suggestive or explicit
- excessively gory:

This includes:
- display of the ejection of internal organs
- excessively detailed depiction of mutilation or maiming`

**If I developed a game using this plugin wherein you could nothing except open a full copy of a King James Bible, I would be in direct violation of your license.** Even if I used other translations, the text is pretty clear about beheadings, circumcision (also known as genital mutilation), and the maiming of various characters.

Not to mention instances such as Sodom and Gomorrah, where God and his angels basically commit war crimes by razing two entire cities filled with men, women, and children. Don't even get me started on the flood, the plagues, blinding people, etc. I personally would consider _drowning countless human children to death_ to be pretty amoral, but then again I'm no divine being. Depicting drowning children isn't prohibited by the license though, so it's wonderful that anyone who makes "Baby Drowning Simulator 2022" could do so without running into any moral/legal trouble from your standpoint.

Let's also not forget that the Bible says (in the New Testament, even!) that slaves should obey their masters, tells slave owners essentially "be nice to your slaves," and provides guidelines for slave trading - which I would assume goes against the "slavery" portion of the license, but if not, then it certainly would violate the "human trafficking" and "indentured servitude" clauses. 

And while I'm not going to get into the details - there's no biblical law or guidance regarding abortion. That's an entirely modern aspect of Christian society/politics, and its inclusion in this license stems from your own personal tenets rather than from the religious text you claim to be invoking here.

The entire notion of this license directly violates the ideals Jesus is typically interpreted as providing in the Sermon on the Mount. For example, the entire virtue of "turning the other cheek" is that you aren't responsible for correcting the sins of others - rather it's God's judgement that will ultimately punish wrongdoing and evil intentions. In other words, Jesus has granted you immunity from sins committed or immoralities depicted by other people using your work. 

To quote Jesus, "Render therefore unto Caesar the things which are Caesar's; and unto God the things that are God's." As solid as I assume your code is in the plugin, I really hope you don't claim that it's _divine_. We're approaching TempleOS levels of delusion if that's the case, and I very desperately hope that it isn't.

Anyways, if you're in the mindset of "it's my work and I can put whatever rules I want in it" then sure - that's entirely valid - but don't try to hide your own personal intentions behind an "objectively true and good" religious text when there's clearly either strong personal bias or ulterior motives behind the licensure. Please consider using a license that's founded on reality, or at least one that isn't so arbitrarily based on your own opinions.

**Comment by opax7** [comment]
To continue the thought, “the issue is obviously closed”, I think we can all agree that this issue will never truly be closed. We, the creators of the Katharos License, were fully aware that this license would cause controversy and that assuredly there would be those who would be disappointed with its implementation. For us however, the alternative to the Katharos License was to abandon the project; a point that has already been made. So our decision was to disappoint some as opposed to neglecting all.

I wanted to respond here because I can sense the passion that you have for your view point and wanted to address your statements with my own passion.

First of all I want to address the kindness of the others in response to the Katharos License, and your lack of respect for their choice to discuss a controversial matter with decorum and civility. Just because we disagree on matters and are passionate about our viewpoints, if we cannot discuss them openly and freely without hostility, the value of such discussions diminish quickly. And while you may think that our ideas are “stupid” (we are all entitled an opinion), they most certainly are not thoughtless. With that said, I would encourage you to take a more tactful approach to any debate so that your point may be well received and not thought to be rants of belligerence.

With that out of the way, let’s talk about objective “good” and how it relates to deriving legal documentation. I find it a bit ironic how you negate validity of using the fact that the Bible can be interpreted in many different ways, as a reason that it can’t or shouldn’t be used to derive a legal document. The reality is that law, and the execution of law is highly dependent on interpretation, which is why it is typically so verbose. But law is also based on some foundation that underpins the values that the law seeks to uphold. It is interesting that when laws, or in this case a license, states that its underpinning values are derived from the Bible, it is preposterous. But the laws of free countries are also underpinned by some document that has been chosen to uphold the founding principles. 

But there must be some genesis of morality, or some basic set of principles that are used to base the derived rules of law. Even if that set of principles is the absence of all morality. In that case there is still a principle to base all other rules off of. Kind of like a no-holds-barred fight match where the only rule is there are no rules. There is still “one” rule that all others are based on. We’ve only chosen to use the Bible (yes, highly up to interpretation) as our genesis of morality. 

You mentioned that we are trying “to hide your own personal intentions behind an ‘objectively true and good’ religious text when there’s clearly either strong personal bias or ulterior motives behind the licensure.” And that we should “consider using a license that’s founded on reality.” We are not hiding behind our motives at all. Our motives are quite clear and we are very open about them. And for clarity I will state simply that we seek to only get behind anything that would bring glory to God and forego anything that does not. We believe that the Bible is the standard for us to do that. There is nothing fake or coercive about this, our stance, or our motives. 

You also stated, “To claim you understand any ‘objective’ moral good from the Bible is to claim that you can somehow invoke God’s will directly, or that you perfectly understand His intent when divinely inspiring the text.” Do I think I understand the will of God. Of course I do. That is why the Lord blessed us with the Bible, as well as an interpreter named the Holy Spirit. Do I think I always get it right? Absolutely not. I’m not perfect, nor do I claim to be. But it also doesn’t take a theologist to understand some of the basic facets of the moral compass provided by the Bible. And even the theologists get things wrong. They are just sinful man as well. And while this is not the forum to discuss Bible apologetics, I would be more than happy to discuss with you offline if you would like to debate all the millions of things we could debate.

You seem to be inferring that if something is mentioned in the Bible, then it gets a free pass in the terms of the license, and if it is not in the Bible then the license has nothing to say about that issue. But the license itself stands alone. You mentioned that you are aware of this fact, yet you go on to mention happenings in the Bible and mingle them with violations or allowances of the license. It is this mingling that we need to avoid. The Bible is what guides our principles and that has provided the motivation for the license. Just like corporate vision or indeed general public courtesy that provides motivation for establishing codes of conduct for meetings, professional gatherings, or online forums. 

There are many other things you mention about your own interpretation of the Bible which again, we can debate in an appropriate forum if you like. But to distill your comment down to its basic principle is to say that it simply “sucks” that you won’t use the tool. Not because the license is too restrictive, but because its motivation comes from the Bible. Or more specifically, our interpretation of the Bible. I would encourage you to submit an issue that focuses more on the license text itself if there is something specific in there that particularly rubs you the wrong way or prevents you from using it. In such an instance we could at least perform some analysis on whether the license needs to change to facilitate or whether we simply won’t budge on that particular aspect of the license. We want to be accommodating to as many as possible, while upholding our values. I think that’s fair.

**Comment by azbay** [comment]
> Just because we disagree on matters and are passionate about our viewpoints, if we cannot discuss them openly and freely without hostility, the value of such discussions diminish quickly. And while you may think that our ideas are “stupid” (we are all entitled an opinion), they most certainly are not thoughtless. With that said, I would encourage you to take a more tactful approach to any debate so that your point may be well received and not thought to be rants of belligerence.

It's just that when I see a group of adults who are 1) intelligent to a great degree, as is implied by the accomplishments herein and your overall diction, but 2) have been deluded by lies and false promises - it's incredibly frustrating and sad to me. Don't think that any vitriol that I put forth is directed at you or anyone in your company. It's directed at those institutions and texts that have lied to and indoctrinated you so fully.

> You seem to be inferring that if something is mentioned in the Bible, then it gets a free pass in the terms of the license, and if it is not in the Bible then the license has nothing to say about that issue. But the license itself stands alone.

Then divorce the license from the Bible entirely, including the text in the preamble. Then at least you'll be consistent, and less inflammatory - that is, the license will be perceived as being influenced by your own morals, biases, and prudery rather than some invocation of divine "objective" morality. If the license stands on its own then there's no need for the association at all, especially when it so blatantly contradicts the book it claims to derive its clauses from (which in itself is a contradictory text on a number of levels - not a good combination).

On the topic of concerns about the restrictions of the license itself, I believe others have listed a number of reasons that there are issues with them. To reiterate some of these concerns and possibly add to them, though:

- Clause 6.1 (in the revised 2.0 draft, assuming this is the "latest" version) should be eliminated entirely or be rewritten from scratch. It's vague, limits artistic freedom, and no one creating something with actual characters would take the risk of using an asset with this license attached. "Sexually suggestive" is particularly frightening, given how easily that can be interpreted as _anything, really._ If you want to include something along the lines of "should not be used in products that include sexual nudity or pornography" then that'd be more reasonable. 
- In the other concerning bullet point of the same section, regarding "excessively gory" content - that too is incredibly vague. The 2.0 draft attempted to quell concerns about the vagueness but gives no real comfort, as "excessively detailed depiction of mutilation or maiming" is up to interpretation of what "excessively" means exactly, what kind of mutilation we're talking about (there are two main definitions, one involving the removing of someone's body parts, the other involving "inflicting severe damage to something" which could apply to inanimate objects), and maiming, which is featured in many games when injuries are inflicted (even Super Mario Bros. could be considered to contain maiming - Goombas damaging Mario can cause him to lose abilities and even shrink in size, effectively maiming his movement - though whether it's "excessively detailed" is wholly up to interpretation).

To wrap up, beyond all of the things you chose to ignore about my first post, I can't say I'm unsurprised you neglected the portion regarding the need for this license in the first place, but I am a bit disappointed. Particularly regarding the notion that if someone else uses something you've created to make another product that's "impure" from your standpoint, that it'd somehow be a moral failing on your part or that you'd be somehow held in lesser regard by your "heavenly father." If I were in your shoes (and at one point in my life I was, at least in the vigor of my belief and my desire to implement my faith into my creative passions) I would simply ensure that my product wasn't explicitly made for the purpose of creating immoral products, or including features myself that could be seen as immoral or sinful. My neighbor's business is his own, and his sins are his own - it's for God (and only God alone) to judge the righteousness of another's actions.

Additionally, abiding blindly to certain doctrines, laws, or tenets of the faith without taking into account the intent and the goodness of the individual is inherently sinful. Jesus defends King David's actions in 1 Samuel wherein he eats the holy bread when hungry. While this isn't a life-or-death situation, the principle applies - if you intend to do good, then you can do no wrong by accident, out of providing for others, or by the impure actions or others. If you truly do care about purity and goodness and put that ethos into your work, then that's all that matters in the grand scheme of things.

**Comment by opax7** [comment]
@azbay,

I appreciate your comments here. With your statements regarding the license, we can analyze your concerns and take action accordingly. This is a much more productive step.

And once again, I understand the passion for your viewpoint as we are passionate for ours as well. I respect your viewpoint, and wish that ours wasn’t so frustrating and sad to others. But that is the nature of having differing opinions. As mentioned, we will analyze your issues with the license and take action as we deem necessary. 

With respect,

@opax7

**Comment by zicklag** [comment]
> To wrap up, beyond all of the things you chose to ignore about my first post, I can't say I'm unsurprised you neglected the portion regarding the need for this license in the first place, but I am a bit disappointed. Particularly regarding the notion that if someone else uses something you've created to make another product that's "impure" from your standpoint, that it'd somehow be a moral failing on your part or that you'd be somehow held in lesser regard by your "heavenly father."

This is something that could possibly be made more clear in the preamble. In making the Katharos License and licensing our tools under it, we don't condemn anybody else who makes things and doesn't restrict what people are allowed to do with it. We feel that God has put it on _our_ consciences that people will use what we make, specifically for the video-game market, to design things that contradict the very reason we created our tools in the first place, which is to share good and pure things ( by our interpretation of God's will ) with the world.

It is purely something we feel put on our hearts. We don't say that God condemns everybody who makes something that somebody else uses for evil. That is not at all what we mean to imply with the Katharos License. We just feel that God called **us** to take this stance with **our** work in this field which is so fraught with things that are not in agreement with our convictions.

---

When it comes to vagueness of terms, that is something we will attempt to address. We recognize that making those terms more clear is important and we agree that we should clarify those terms as much as possible.

---

Finally, coming back to this discussion actually brought quite an interesting new idea to mind that we had not considered before, regarding the lack of case law and some related legal concerns with the license. 

@alice-i-cecile mentioned above:

> It induces very complex legal risk due to challenges in definition and lack of case law

This has been a challenge with the license that we had no real solution for. The fact of the matter is that the license is atypical and courts may choose to refuse to interpret the license entirely or otherwise rule in an unpredictable way due to the lack of precedent.

Hopefully this latest idea provides at least some help in that direction. I've created an issue on the main Katharos License repo to outline the idea and open up any discussion if you are interested in discussing the more technical challenges of the license: https://github.com/katharostech/katharos-license/issues/3.

**Comment by Foobin** [comment]
As someone who has no religious beliefs but grew up with devoutly religious family members and friends, I have very strong feelings about how the license has been written.  I have to admit that it made me quite angry.

To me, the license reads as a group of individuals pushing their **own beliefs**, and using the word of God as justification for their ideas. What they have written are their owns words and their own interpretations, not God's. They could just have easily have written a clause barring the depictions of homosexual relationships. 

To say that these "truths" are the truths of God is insulting. There are Christians who support abortion. There are Christians who support healthy engagement with sexuality. To them these beliefs do not exist in contradiction with the texts.

I suggest that you alter the preamble to say the clauses imposed by the License constitute YOUR interpretation of the Bible and that that interpretation does not necessarily reflect God's will. It is after all, only YOUR interpretation.

Thank you.

**Comment by meisterpeeps** [comment]
It seems to me that your crate has a lot of neat features but your license is bad-- that is, the antithesis of "good".   It makes many errors and has many flaws, most egregiously in the preamble.  

For example, it states:  

"In order to maintain objectivity in a highly subjective matter, there must be some source of truth from which to derive said objectivity. The source of "truth" for the Katharos License, and where the definition of what is "good" and "pure", come from the Word of God, The Holy Bible."

The Christian bible was neither created nor inspired by a god or gods.  It was created by many different humans, and too often inspired by vice, vanity, and evil rather than anything virtuous. The source of truth to maintain objectivity on what is good can only be found in a shared rationality and moral sense that most humans possess and can cultivate.  The Christian bible is unfortunately not a good source for developing rationality or a moral sense; while people can still develop strong rational skills and a virtuous moral sensibility despite having irrational faith in the Christian bible, it can be easily observed that too often it fails to bring forth virtue in the human soul or even leads to spiritual degeneration.

"Kill every male among the little ones, and every woman who has known a man by lying with him, but those women who have not known a man by lying with him, you may keep alive for yourselves”. Numbers 31:17

There is an abundance of gory and violent imagery of abortion and murder in the Christian bible -- and approved by the main protagonists.   In that line alone from Numbers, the character of Moses, a man glorified in the bible, sanctions war crimes, murder, sexual slavery, and the slaying of pregnant women -- abortion.   If someone were to create a game depicting the bible using your crate, it would be a violation of the crate's license.  Yet you claim to base the moral foundations of the license on the bible itself.  

Your license is self imploding.   It's bad.  And I'll bet my socks that if there are any rational, virtuous gods, they agree with me.

**Comment by zicklag** [comment]
I've just created a list of [frequently posed challenges](https://github.com/katharostech/katharos-license#frequently-posed-challenges) for our license and our responses that should cover the majority of points.

**Comment by shinaka** [comment]
I unfortunately realize what a waste of time this is likely to be, but I think there’s a bigger issue at hand here that’s worth discussion:
Code does not exist in a vacuum. Hundreds of thousands of closed-source and open-source products have come together to create the current state of computing as we know it. The number of ‘useful’ user-facing programs that can proudly say they rely on ABSOLUTELY nothing else (whether it be libraries, languages, linkers, compilers, etc) is incredibly small.

Here we are, discussing the license of a group of plugins for an open source game engine (open also in the sense of: anyone can use it to make whatever CONTENT they wish to make). And this is true of… most open source licenses. They’ll typically dictate limitations on commercial use, requirements to do so (releasing source code), attribution, etc.

But, here, a group of plugins for an open source game engine, potential users are being told by the author(s): to use this code (and then create actual content), you have to be limited to my religion’s whims on what’s right and wrong. My specific belief system is now your problem for this ‘open source’ plugin library for an open source game engine, built on other numerous open source libraries, written in an IDE using countless open source libraries, on an operating system also making use of said-numerous-countless open source libraries.

Except.. right in that chain of ‘why open source is great’, you decided, if someone uses a small amount of your code in something they create that you don’t like — well now its your problem somehow! I mean think about if each open source piece of software used a license like this, applied their religious beliefs to the application of their code, etc. We’d have nothing. We’d have no open collaboration. At best we’d create… ‘religious sects’ of software.. which.. never thought I’d have to type that phrase!

This is anti open-source in every way. I’d argue its also hypocritical to use whatever libraries you wish without any religious edicts preventing you from doing so… to then do exactly that.

**Comment by BuyMyMojo** [comment]
I would suggest using the [Bigg Butts License+Apache-2.0](https://github.com/BuyMyMojo/big-butts-license) instead as it is more inclusive and protects the rights of users more without relying on a crypto tool for enforcement.
